scripts/logbook.py

```diff
@@ -240,44 +746,18 @@ def main(lims, pid, epp_logger):
             else:
                 instrument_name = pro.instrument.name
                 log.append(instrument_name)
-                if record["details"][instrument_number] is not '':
-                    for item in record["details"][instrument_number]:
-                        try:
-                            udf_content = pro.udf[item[4:]]
-                            udf_detail.append(item[4:]+":"+pro.udf[item[4:]])
-                        except KeyError, e:
-                            continue
-                    if udf_detail == []:
-                        log.append("-")
-                    else:
-                        log.append(','.join(udf_detail))
-                    write_record(log,record["dest_file"][instrument_number])
-                else:
-                    log.append("-")
-                    write_record(log,record["dest_file"][instrument_number])
+                details = get_details(record[instrument],pro)
+                log.append(details)
+                write_record(log,record[instrument]["dest_file"])
 
         elif instrument[0:3] == "udf":
             try:
                 instrument_name = pro.udf[instrument[4:]]
-                if pro.udf[instrument[4:]] is not '':
+                if pro.udf[instrument[4:]] != '':
                     log.append(instrument_name)
-                    if record["details"][instrument_number] is not '':
-                        for item in record["details"][instrument_number]:
-                            try:
-                                udf_content = pro.udf[item[4:]]
-                                udf_detail.append(item[4:]+":"+pro.udf[item[4:]])
-                            except KeyError, e:
-                                continue
-                        if udf_detail == []:
-                            log.append("-")
-                        else:
-                            log.append(','.join(udf_detail))
-                        write_record(log,record["dest_file"][instrument_number])
-                    else:
-                        log.append("-")
-                        write_record(log,record["dest_file"][instrument_number])
-                else:
-                    continue
+                    details = get_details(record[instrument],pro)
+                    log.append(details)
+                    write_record(log,record[instrument]["dest_file"])
```

Comment:
These three line are exactly the same in all of the condition blocks, so maybe you can keep these outside the `if ... elif .. elif..` block

Reply:
Thanks everyone! I will try to do another PR based on your comments ASAP. /chuan